### DISCUSSION TITLE
connection.connector.getChainId is not a function

### DISCUSSION THREAD

**Discussion by Sahitya777** [wevm/wagmi#4216] [Q&A]
So the issue is that I have one base route in my application as /v1 where I have a modal for wallet connects and when the user connects the wallet I send him to other route like /home where the wallet stays connected by connectWagmi({ connector:wagmiConnectors[0] })
But the thing is If i refresh the page the write calls start to fail and give error of connection.connector.getChainId is not a function Idk why it is happening 
This is my config
  const config = createConfig(
    getDefaultConfig({
      // Your dApps chains
      chains: [baseSepolia],
      transports: {
        // RPC URL for each chain
        [baseSepolia.id]: http(
        ),
      },
  
      // Required API Keys
      walletConnectProjectId: process.env.NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID as string,
  
      // Required App Info
      appName: "Your App Name",
  
      // Optional App Info
      appDescription: "Your App Description",
      appUrl: "https://family.co", // your app's url
      appIcon: "https://family.co/logo.png", // your app's icon, no bigger than 1024x1024px (max. 1MB)
    }),
  );

**Comment by mdesousa** [comment]
we are also running into this @Sahitya777 ... looks like a bug. did you create an issue in the wagmi repo? i'd like to upvote it if it's there... thanks.

**Reply by Sahitya777** [reply]
Hey so basically what I found is that you have to use ssr in this and that will do the trick for you, you dont have to reconnect again and again on reload and there is no writecontract error as well so try that out once.
https://wagmi.sh/react/guides/ssr

**Reply by mdesousa** [reply]
ah thanks… but we are using vite so it’s all client-side. we may be running into a different issue then.

**Reply by Yokat15** [reply]
Same issue trying to link teneo.pro
and this "n.connector.getChainId is not a function" keeps popping up, i used 3338.
what should i do please?

**Comment by kundanlmr** [comment]
hey i am also the same issues like r.connector.getChainId is not function
config file
export const config = defaultWagmiConfig({
  chains: [defaultAppChain],
  projectId,
  metadata,
  enableWalletConnect:true,
  enableInjected:true,
  enableEIP6963:true,
  enableCoinbase:true,
  ssr:true,
  storage: createStorage({
    storage: cookieStorage
  })
})
the issues is not face before the same config or write , read contract function was working but now i am getting this error , only  write contract function is giving this error  other write function are working fine with same configuration  this issues is not facing all users only some user getting this error
what could be the reason?

**Reply by Asma1665** [reply]
Ton link

**Reply by Asma1665** [reply]
Ton link chain blockchair

**Comment by nakedfool** [comment]
++++

having same issues as well!

**Comment by Sotatek-AaronDinh** [comment]
Same issue :(

**Comment by caotrongtin99** [comment]
same issue

**Reply by nnn-gif** [reply]
following

**Comment by Adebesin-Cell** [comment]
Has anyone been able to resolve this? @Sahitya777

**Reply by AlphaDio** [reply]
Yes, I have been able to solve this by using the right config object.

I had 2 objects that were playing the role of config, one made with old Wagmi code, and the other made using ReOwn.

I was using the config from Wagmi, when the Wagmi component had the config from ReOwn. This, I guess, was at the origin of the bug.

**Reply by Adebesin-Cell** [reply]
Thanks for your response! 👏 

In my case, I'm using ``getDefaultConfig`` from ``@rainbow-me/rainbowkit`` and then passing it to my wagmi provider which seems accepted in docs https://www.rainbowkit.com/docs/installation.

I'll inspect more if it's double config that's happening in my case, thanks!

**Reply by agualis** [reply]
Hey @Adebesin-Cell, we were having the same issue with rainbowkit but it's not happening anymore (3 weeks ago). Did you update viem + wagmi + rainbowkit to it latest version?

**Reply by Adebesin-Cell** [reply]
Not yet, let me try that, thanks @agualis 👏

**Reply by enesozturk** [reply]
@AlphaDio @Adebesin-Cell This is Enes from [AppKit](https://github.com/reown-com/appkit) (aka web3modal) - saw both of you resolved the issue, if you still encounter the issue let me know please

**Comment by AlphaDio** [comment]
It seems to come from when there is a divergence between the config used by the Wagmi wrapper component, and the config used by transactions (readContract, writeContract).

**Comment by gte620v** [comment]
I had this issue but it seems to be due to the browser wallet silently signing me out of my account, not a wagmi misconfiguration.  Once the sign out happens, wagmi starts throwing this error.  I suspect that metamask just changed their sign-out delay and now this is happening to many users.

**Comment by alejandro99so** [comment]
I had the same error and I solved it with the following code:

```
import { useConnections, useSignMessage } from "wagmi";
const connections = useConnections();
useEffect(()=> {
    if (myConditional > 0 && connections.length) {
      signMessage({...});
    }
  }, [myConditional, connections]);
```
  
  U have to wait until connection will be ready in order to sign messages or use custom hooks

**Comment by Royal-lobster** [comment]
For me, the wagmi config used to be inside the provider component. Moving it outside fixed this

**Comment by shayesteMalekpour** [comment]
any update?

**Reply by pho2345** [reply]
try to with my comment

**Comment by pho2345** [comment]
``` typescript
import { http, defineChain } from 'viem'
import { createConfig } from 'wagmi'
import { bsc, bscTestnet } from 'wagmi/chains'

//custom define Chain
const bscTest = defineChain({
  id: 97,
  name: 'BNB Smart Chain Testnet',
  nativeCurrency: { name: 'BNB', symbol: 'BNB', decimals: 18 },
  rpcUrls: {
    default: {
      http: ['https://bsc-testnet-dataseed.bnbchain.org/'],
    },
  },
  blockExplorers: {
    default: {
      name: 'BNB Smart Chain Testnet',
      url: 'https://testnet.bscscan.com/',
      apiUrl: 'https://api-testnet.bscscan.com/api',
    },
  },
})

export const config = createConfig({
  chains: [
  bscTest,
    bscTestnet,
    {
      ...bsc,
      rpcUrls: {
        default: { http: ['https://bsc-dataseed3.bnbchain.org/'] },
      },
    },
  ],
  transports: {
   [bscTest.id]: http(),
    [bscTestnet.id]: http(),
    [bsc.id]: http(),
  },
  ssr: true,
})

<WagmiWeb3ConfigProvider  config={config}  chains={[bsc, bscTestnet, bscTest]} .... />


```
i solve it with chain from 'wagmi/chains' , you must put chains  in provider to not got error "Not found chain"

**Comment by jgresham** [comment]
Having this issue too on a vite client-only (ssr false) setup

**Reply by jgresham** [reply]
For me, I was passing a different wagmi `config` to my provider from the `config2` that I was passing to wagmi's `signMessage`

**Comment by Missfabl** [comment]
Please I'm having issue connecting my wallet to EVM wallet, can it be resolved immediately because it's urgent for me

**Comment by blmalone** [comment]
+1 still seeing this in 2026.

**Comment by lucadonnoh** [comment]
same issue

**Comment by mrwillis** [comment]
Still seeing this unfortunately

Likely related

https://github.com/wevm/wagmi/issues/3589
https://github.com/wevm/wagmi/discussions/3834
